src/reducers/featureReducer.js

```diff
@@ -0,0 +1,48 @@
+import { ADD_FEATURE, REMOVE_FEATURE, UPDATE_TOTAL } from '../actions/featureAction';
+
+const initialState = {
+    additionalPrice: 0,
+    car: {
+      price: 26395,
+      name: '2019 Ford Mustang',
+      image:
+        'https://cdn.motor1.com/images/mgl/0AN2V/s1/2019-ford-mustang-bullitt.jpg',
+      features: []
+    },
+    additionalFeatures: [
+      { id: 1, name: 'V-6 engine', price: 1500 },
+      { id: 2, name: 'Racing detail package', price: 1500 },
+      { id: 3, name: 'Premium sound system', price: 500 },
+      { id: 4, name: 'Rear spoiler', price: 250 }
+    ]
+  };
+
+export const featureReducer = (state=initialState, action) => {
+    switch (action.type) {
+        case ADD_FEATURE:
+            return {
+                ...state,
+                car: {
+                  ...state.car,
+                  features: state.car.features.includes(action.payload) ? [...state.car.features] : [
+                    ...state.car.features, action.payload
+                  ]
+                }
+            }
+        case REMOVE_FEATURE:
+            return {
+              ...state,
+                car: {
+                  ...state.car,
+                  features: state.car.features.filter(c => c.id != action.payload.id)
+                }
+            };
+        case UPDATE_TOTAL:
```

Comment:
So I see it the prices being added on, but how are we removing the prices from the updated?